### DISCUSSION TITLE
.NET 8 lack of symbols *.ni.pdb

### DISCUSSION THREAD

**Discussion by nietras** [dotnet/runtime#96897] [Q&A]
I am having issues with .NET 8 (SDK 8.0.101) on Windows in Visual Studio 2022 and no symbols being loaded for many dlls. This means I lack info when e.g. profiling etc. I know this is known issue from back in the old days with .NET Framework and NGEN, but I thought this had been solved and was usable in Visual Studio.
```
Symbols failed to load for system.private.corelib.dll. Failed to find symbol file System.Private.CoreLib.ni.pdb
Symbols failed to load for system.diagnostics.process.dll. Failed to find symbol file System.Diagnostics.Process.ni.pdb
Symbols failed to load for system.console.dll. Failed to find symbol file System.Console.ni.pdb
```
I can't remember or find the information on how to get symbols for these, so my question is how can get this working?

**Comment by am11** [comment]
Usually VS downloads them from Microsoft SymStore when 'Just My Code' is disabled, but there is also a command-line tool for manual downloading: https://learn.microsoft.com/en-us/dotnet/core/diagnostics/dotnet-symbol

Powershell / Run as Administrator:
```pwsh
> dotnet tool install --global dotnet-symbol
> dotnet-symbol "$(((gcm dotnet).Source) | split-path -parent)\shared\*.dll" --recurse-subdirectories --timeout 20
```

**Reply by nietras** [reply]
Disabling `Just My Code` did not change anything. The `dotnet-symbol` does not appear to be able to find symbols for `*.ni.dll` that R2R files e.g.:
```
ERROR: Not Found: System.Runtime.Numerics.ni.pdb - 'https://msdl.microsoft.com/download/symbols/system.runtime.numerics.ni.pdb/737c671f4ac6fa84a05a432fe920e2c41/system.runtime.numerics.ni.pdb'
Writing: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.1\System.Runtime.Numerics.pdb
````
Based on using:
```
dotnet-symbol "$(((gcm dotnet).Source) | split-path -parent)\shared\Microsoft.NETCore.App\8.0.1\*.dll" --recurse-subdirectories --timeout 20
```
to download only for .NET 8.0.1. Hence, this did not appear to resolve the issue.

**Comment by MichalPetryka** [comment]
Don't you need to set
```xml
<PublishReadyToRunEmitSymbols>true</PublishReadyToRunEmitSymbols>
```
to get R2R symbols?

**Comment by jkotas** [comment]
> Failed to find symbol file System.Private.CoreLib.ni.pdb

This is .NET 8 bug, tracked by https://github.com/dotnet/runtime/issues/96917 and related issues. It will be fixed in future servicing update.

**Comment by HighPerfDotNet** [comment]
Run into a lot of missing symbols too (attempted download for 8.0.1 as explained above), specifically was after System.Net.Http.ni.pdb - this and lots of other .ni.'s are also missing from 8.0.0 symbols :(
